tests/unit/test_matrix.py

```diff
@@ -152,8 +159,50 @@ def it_knows_how_to_extract_counts_for_matrix_creation(
             == expected
         )
 
+    @pytest.mark.parametrize(
+        ("dim_types", "matrix_type", "parent_type"),
+        (
+            ((DT.CAT, DT.CAT), _CatXCatMatrix, _BaseCubeResultMatrix),
+            ((DT.MR, DT.CAT, DT.CAT), _CatXCatMatrix, _BaseCubeResultMatrix),
+            ((DT.MR, DT.MR), _MrXMrMatrix, _CatXCatMatrix),
+            ((DT.MR, DT.CAT), _MrXCatMatrix, _CatXCatMatrix),
+            ((DT.CAT, DT.MR, DT.CAT), _MrXCatMatrix, _CatXCatMatrix),
+            ((DT.CAT, DT.MR), _CatXMrMatrix, _CatXCatMatrix),
+            ((DT.CAT, DT.CAT, DT.MR), _CatXMrMatrix, _CatXCatMatrix),
+            ((DT.MR, DT.CAT, DT.MR), _CatXMrMatrix, _CatXCatMatrix),
+        ),
+    )
+    def it_can_create_regular_matrix(
```

Comment:
Change this to make more sense according to our discussion with @scanny , and according to the changes in the comments.